Automattic/jetpack
# Change style used for JS dependencies

#### Changes proposed in this Pull Request:

It's annoying to have to update all the dependencies of monorepo JS
packages for every patch-level version bump. Let's make it work like the
PHP dependencies, where the dependencies are only updated on minor and
major bumps.

The new scheme looks like `workspace:* || ^1.2`. Within the monorepo the
`workspace:*` part is operative and the `^1.2` is ignored by pnpm but
will still be checked by CI.

When we publish we can strip the `workspace:* || ` bit leaving only the
`^1.2` for external users. But I'll save that change for a followup PR.

#### Jetpack product discussion


p9dueE-4NR-p2#comment-7660

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* `jetpack changelog add` a patch-level change for js-packages/base-config. Then run `tools/fixup-project-versions.sh`.
  * On master, that should result in a bunch of other deps being bumped. And then their deps too if they're not already on an `-alpha` version, etc.
  * With this PR, only base-styles itself should be changed.
* `jetpack changelog add` a minor or major change for js-packages/base-config. Then run `tools/fixup-project-versions.sh`.
  * With this PR, all the deps on `@automattic/jetpack-base-styles` should get the version bump to `workspace:* || ^0.2`. But no other deps should be bumped.
